Types.d.ts

```diff
@@ -0,0 +1,36 @@
+export type GenericMeta =
+  {
+    // ???
+  };
+
+export type BareHeaders = Map<string, string | string[]>;
+
+export type SmallResponse =
+  {
+    body: ReadableStream | ArrayBuffer | Blob | string,
+    headers: BareHeaders,
+    status: number
+  }
```

Comment:
Personally, I would yoink the prototype interface from lib.dom.d.ts, remove the readonly keywords, and make some properties optional